.github/workflows/ci.yml

```diff
@@ -0,0 +1,27 @@
+name: run tests 
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+    unit-tests: 
+      runs-on: ubuntu-latest
+
+      steps: 
+        - name: Checkout code
+          uses: actions/checkout@v4
+
+        - name: Set up environment 
```

Comment:
pnpm is a faster replacement for npm, see e.g. this workflow: https://github.com/spatial-model-editor/spatial-model-editor.github.io/blob/3d93eac74e8f2b0e1b29429cb5c849602bb19093/.github/workflows/deploy.yml#L11-L20

Reply:
:+1: